app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/NrsServiceNew.scala

```diff
@@ -0,0 +1,129 @@
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
+import uk.gov.hmrc.excisemovementcontrolsystemapi.connectors.NrsConnectorNew
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.auth.ParsedXmlRequest
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.nrs._
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.NRSWorkItemRepository
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.model.NrsSubmissionWorkItem
+import uk.gov.hmrc.excisemovementcontrolsystemapi.services.NrsService.nonRepudiationIdentityRetrievals
+import uk.gov.hmrc.excisemovementcontrolsystemapi.utils.{DateTimeService, EmcsUtils, NrsEventIdMapper}
+import uk.gov.hmrc.http.{Authorization, HeaderCarrier, InternalServerException}
+import uk.gov.hmrc.mongo.workitem.{ProcessingStatus, WorkItem}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class NrsServiceNew @Inject() (
+  override val authConnector: AuthConnector,
+  nrsConnectorNew: NrsConnectorNew,
+  nrsWorkItemRepository: NRSWorkItemRepository,
+  dateTimeService: DateTimeService,
+  emcsUtils: EmcsUtils,
+  nrsEventIdMapper: NrsEventIdMapper,
+  correlationIdService: CorrelationIdService
+)(implicit ec: ExecutionContext)
+    extends AuthorisedFunctions
+    with Logging {
+
+  def makeNrsWorkItemAndAddToRepository(
+    request: ParsedXmlRequest[_],
+    authorisedErn: String,
+    correlationId: String
+  )(implicit headerCarrier: HeaderCarrier): Future[Done] = {
+
+    val payload        = request.body.toString
+    val userHeaderData = request.headersAsMap
+    val message        = request.ieMessage
+    val exciseNumber   = authorisedErn
+    val notableEventId = nrsEventIdMapper.mapMessageToEventId(message)
+
+    for {
+      identityData  <- retrieveIdentityData()
+      userAuthToken  = retrieveUserAuthToken(headerCarrier)
+      metaData       = NrsMetadata.create(
+                         payload,
+                         emcsUtils,
+                         notableEventId,
+                         identityData,
+                         dateTimeService.timestamp().toString,
+                         userAuthToken,
+                         userHeaderData,
+                         exciseNumber
+                       )
+      encodedPayload = emcsUtils.encode(payload)
+      _             <- nrsWorkItemRepository.pushNew(NrsSubmissionWorkItem(NrsPayload(encodedPayload, metaData)))
+    } yield Done
+  }
+
+  def submitNrs(workItem: WorkItem[NrsSubmissionWorkItem])(implicit hc: HeaderCarrier): Future[Done] =
+    //needs to call connector, and handle marking the workitems. IN PROGRESS
+    for {
+      _ <- nrsConnectorNew.sendToNrs(workItem.item.payload, correlationIdService.generateCorrelationId())
+      _ <- nrsWorkItemRepository.complete(workItem.id, ProcessingStatus.Succeeded)
+    } yield Done
+
+  private def retrieveIdentityData()(implicit headerCarrier: HeaderCarrier): Future[IdentityData] =
```

Comment:
Maybe I'm overthinking this, but part of me thinks this could benefit from being in some sort of tested importable service, or even added to the request as part of an action on the way in (for the two submission endpoints).